### PR TITLE
Added support for vertical connectors (mixed connections)

### PR DESCRIPTION
**Pull request by miroiu**
### 📝 Description of the Change

Added support for vertical connections to address #98 

image

### 🐛 Possible Drawbacks

Increased connection drawing complexity.

### CODE REVIEW

**Comment by hendrikp** [comment]
Yes, thats pretty much how i did it, i just split it to `TargetVertical`, `SourceVertical` e.g. if done through your enum maybe `TargetOrientation` and `SourceOrientation`. Then i  differentiated between source/target point for the calculation and used them where appropriate (thus allowing flow form one orientation to another). If you want i can submit an extension to this PR. in that regard.

**Comment by miroiu** [comment]
That would be awesome! I'm curious if you implemented this for all the existing connections or only for the default one, and if they still work like before for horizontal `SourceOrientation` and `TargetOrientation`.

**Comment by hendrikp** [comment]
hm yeah ill make sure they work the same - the default one definitely did, i tested on the bezier one, the line one should be easy, ill have to see about circuit.

**Comment by hendrikp** [comment]
See #100 had to create a stacked PR due to rights.

File: Nodify/Connections/BaseConnection.cs [hunk 1]
```diff
@@ -118,6 +118,7 @@ public abstract class BaseConnection : Shape
         public static readonly DependencyProperty SourceOffsetModeProperty = DependencyProperty.Register(nameof(SourceOffsetMode), typeof(ConnectionOffsetMode), typeof(BaseConnection), new FrameworkPropertyMetadata(ConnectionOffsetMode.Static, FrameworkPropertyMetadataOptions.AffectsRender));
         public static readonly DependencyProperty TargetOffsetModeProperty = DependencyProperty.Register(nameof(TargetOffsetMode), typeof(ConnectionOffsetMode), typeof(BaseConnection), new FrameworkPropertyMetadata(ConnectionOffsetMode.Static, FrameworkPropertyMetadataOptions.AffectsRender));
         public static readonly DependencyProperty DirectionProperty = DependencyProperty.Register(nameof(Direction), typeof(ConnectionDirection), typeof(BaseConnection), new FrameworkPropertyMetadata(default(ConnectionDirection), FrameworkPropertyMetadataOptions.AffectsRender));
+        public static readonly DependencyProperty OrientationProperty = StackPanel.OrientationProperty.AddOwner(typeof(BaseConnection), new FrameworkPropertyMetadata(Orientation.Horizontal));
```
Comment (hendrikp) [inline Nodify/Connections/BaseConnection.cs:121]: As discussed i recommend the split for more flexiblity

**Review by hendrikp — CHANGES_REQUESTED** [review]
See #100 for extension / review findings. I came to same conclusion, similar implentation but used a bool indication Vertical, but Orientatino is clearer.

File: Nodify/Connections/BaseConnection.cs (hunk 1 above)
Comment (miroiu) [reply Nodify/Connections/BaseConnection.cs]: Thanks for implementing this. We can now draw all kinds of combinations:

image

Comment (hendrikp) [reply Nodify/Connections/BaseConnection.cs]: im kind of curios how you placed your start/end here so it aligns beneath each other smoothly, i based my xaml on the playground and i had to create a attached property which modifies the header/footer port padding, and a visibility binding on those two ports to remove the title.

Did you create a kind of redundant NodeInput/NodeOutput copy of the node class/style to achieve this? Or was there a simpler way avoiding the copy + what i described.

Comment (miroiu) [reply Nodify/Connections/BaseConnection.cs]: I used a simple connector which does not have any padding:

```xaml
<nodify:Node.HeaderTemplate>
    <DataTemplate>
        <nodify:Connector Anchor="{Binding Anchor, Mode=OneWayToSource}" IsConnected="{Binding IsConnected}" />
    </DataTemplate>
</nodify:Node.HeaderTemplate>
```

However, the default `Connector` style does not have a hover effect. You may also want to include a trigger for `IsMouseOver` in the `ControlTemplate.Triggers` collection:

```xaml
<Trigger Property="IsMouseOver"
         Value="True">
    <Setter TargetName="Connector"
            Property="Ellipse.Fill"
            Value="{Binding BorderBrush, RelativeSource={RelativeSource TemplatedParent}}" />
</Trigger>
```

**Pull request merged**